### PR TITLE
fix dbarts extension in R v4.0.0 easyconfigs for non-x86_64

### LINKED ISSUE #11449: dbarts extension added to R-4.0.0-foss2020a breaks on Arm64
https://github.com/easybuilders/easybuild-easyconfigs/commit/858f50f14f5eb7d7719b03594241955be2b5d7d2#diff-f2a7ccd2712a57d5dff6f0bd478d7d18 (via ##11430)) breaks on arm: https://gist.github.com/terjekv/1c98b51b2f5b3b85ef33e3be08c7f3f5
```
simd.c:36:33: error: static declaration of misc_simd_getMaxSIMDInstructionSet follows non-static declaration
   36 | static misc_simd_instructionSet misc_simd_getMaxSIMDInstructionSet(void)
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from simd.c:9:
../include/misc/simd.h:30:26: note: previous declaration of misc_simd_getMaxSIMDInstructionSet was here
   30 | misc_simd_instructionSet misc_simd_getMaxSIMDInstructionSet(void);
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
simd.c: In function misc_simd_getMaxSIMDInstructionSet:
simd.c:37:3: error: expected declaration specifiers before return
   37 |   return MISC_INST_C;
      |   ^~~~~~
simd.c:38:1: error: expected declaration specifiers before } token
   38 | }
      | ^
simd.c:194:17: error: storage class specified for parameter misc_partitionRange
  194 | extern size_t (*misc_partitionRange)(const misc_xint_t* restrict x, misc_xint_t cut, misc_size_t* restrict indices, misc_size_t length);
      |                 ^~~~~~~~~~~~~~~~~~~
simd.c:195:17: error: storage class specified for parameter misc_partitionIndices
  195 | extern size_t (*misc_partitionIndices)(const misc_xint_t* restrict x, misc_xint_t cut, misc_size_t* restrict indices, misc_size_t length);
      |                 ^~~~~~~~~~~~~~~~~~~~~
simd.c:197:15: error: storage class specified for parameter misc_addVectors
  197 | extern void (*misc_addVectors)(const double* restrict x, misc_size_t length, double alpha, const double* restrict y, double* restrict z);
      |               ^~~~~~~~~~~~~~~
simd.c:198:15: error: storage class specified for parameter misc_setVectorToConstant
  198 | extern void (*misc_setVectorToConstant)(double* x, size_t length, double alpha);
      |               ^~~~~~~~~~~~~~~~~~~~~~~~
simd.c:225:15: error: storage class specified for parameter misc_partitionRange_c
  225 | extern size_t misc_partitionRange_c(const misc_xint_t* restrict x, misc_xint_t cut, misc_size_t* restrict indices, misc_size_t length);
      |               ^~~~~~~~~~~~~~~~~~~~~
simd.c:226:15: error: storage class specified for parameter misc_partitionIndices_c
  226 | extern size_t misc_partitionIndices_c(const misc_xint_t* restrict x, misc_xint_t cut, misc_size_t* restrict indices, misc_size_t length);
      |               ^~~~~~~~~~~~~~~~~~~~~~~
simd.c:228:13: error: storage class specified for parameter misc_addVectors_c
  228 | extern void misc_addVectors_c(const double* restrict x, size_t length, double alpha, const double* restrict y, double* restrict z);
      |             ^~~~~~~~~~~~~~~~~
simd.c:229:13: error: storage class specified for parameter misc_setVectorToConstant_c
  229 | extern void misc_setVectorToConstant_c(double* x, size_t length, double alpha);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~
simd.c:231:27: error: expected =, ,, ;, asm or __attribute__ before { token
  231 | void misc_simd_init(void) {
      |                           ^
simd.c:237:13: error: storage class specified for parameter misc_stat_setSIMDInstructionSet
  237 | extern void misc_stat_setSIMDInstructionSet(misc_simd_instructionSet i);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
simd.c:240:1: error: expected =, ,, ;, asm or __attribute__ before { token
  240 | {
      | ^
simd.c:36:33: error: old-style parameter declarations in prototyped function definition
   36 | static misc_simd_instructionSet misc_simd_getMaxSIMDInstructionSet(void)
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
simd.c:296: error: expected { at end of input
  296 | 
      | 
```

### PR DESCRIPTION
**Pull request by terjekv**
(created using `eb --new-pr`)

fixes #11449

### CODE REVIEW

File: easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb [hunk 1]
```diff
@@ -2125,15 +2125,13 @@ exts_list = [
...
-        'checksums': ['15a9c7f2930e2ed3f4c5bcd9b042884ea580d2b2e52e1c68041600c196046aba'],
         'patches': ['liquidSVM-1.2.4-fix_ppc_and_aarch64_build.patch'],
+        'preinstallopts': 'LIQUIDSVM_TARGET="empty"',
         'checksums': [
-            '15a9c7f2930e2ed3f4c5bcd9b042884ea580d2b2e52e1c68041600c196046aba',
+            '15a9c7f2930e2ed3f4c5bcd9b042884ea580d2b2e52e1c68041600c196046aba',  # liquidSVM_1.2.4.tar.gz
             # liquidSVM-1.2.4-fix_ppc_and_aarch64_build.patch
             '46b09e441c3b59af535f20d8db0dee7f1d6a7ddd511175d252115b53cb8b86f8',
         ],
-        # Don't add optimization flags by liquidSVM which may not be known e.g. on PPC
```
Comment (boegel) [inline easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb:2135]: @terjekv Can you restore the comment?

File: easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb [hunk 2]
```diff
@@ -2247,8 +2245,7 @@ exts_list = [
         'checksums': ['4152a1c3c652979e97870e5c50c45a243d0ad8d4ff968091160e3d66509f61db'],
     }),
     ('rda', '1.0.2-2.1', {
-        'checksums': [('6918b62f51252b57f2c05b99debef6136b370f594dc3ae6466268e4c35578ef8',
-                       'eea3a51a2e132a023146bfbc0c384f5373eb3ea2b61743d7658be86a5b04949e')],
```
Comment (boegel) [inline easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb:2251]: Please keep both (alternate) checksums, this is the fault of `--inject-checksums`...

File: easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb [hunk 3]
```diff
@@ -2419,8 +2416,7 @@ exts_list = [
         'checksums': ['3ad01f82132bf75960ae0d8a81cae84eaf4a9ab262f183fc3d6439189e4a3aed'],
     }),
     ('rstantools', '2.0.0', {
-        'checksums': [('4ae78aa39c3d4897f453e621a09961d6f2cda27a1b5657d4d4b32d3d89bd80c6',
-                       'd200a4d8c62d8577fdba819bf770e7abb11c57c3332f4498e1d30ce824598b3a')],
```
Comment (boegel) [inline easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb:2423]: Please keep both alternate checksums

File: easybuild/easyconfigs/r/R/dbarts-0.9-18_fix-simd-on-non-x86.patch [hunk 4]
```diff
@@ -0,0 +1,12 @@
+diff -ru dbarts.orig/src/misc/simd.c dbarts/src/misc/simd.c
```
Comment (boegel) [inline easybuild/easyconfigs/r/R/dbarts-0.9-18_fix-simd-on-non-x86.patch:1]: @terjekv Please add a short description of the patch on top + mention author or upstream reference.

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb (hunk 1 above)
Comment (terjekv) [reply easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb:2135]: Oh, no idea how that happened. Fixed.

File: easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb (hunk 2 above)
Comment (terjekv) [reply easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb]: Fixed.

File: easybuild/easyconfigs/r/R/dbarts-0.9-18_fix-simd-on-non-x86.patch (hunk 4 above)
Comment (terjekv) [reply easybuild/easyconfigs/r/R/dbarts-0.9-18_fix-simd-on-non-x86.patch:10]: @zao is magic. Fixed.

File: easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb [hunk 5]
```diff
@@ -2125,15 +2125,14 @@ exts_list = [
...
-        'checksums': ['15a9c7f2930e2ed3f4c5bcd9b042884ea580d2b2e52e1c68041600c196046aba'],
         'patches': ['liquidSVM-1.2.4-fix_ppc_and_aarch64_build.patch'],
+        'preinstallopts': 'LIQUIDSVM_TARGET="empty"',
         'checksums': [
-            '15a9c7f2930e2ed3f4c5bcd9b042884ea580d2b2e52e1c68041600c196046aba',
+            '15a9c7f2930e2ed3f4c5bcd9b042884ea580d2b2e52e1c68041600c196046aba',  # liquidSVM_1.2.4.tar.gz
             # liquidSVM-1.2.4-fix_ppc_and_aarch64_build.patch
             '46b09e441c3b59af535f20d8db0dee7f1d6a7ddd511175d252115b53cb8b86f8',
         ],
         # Don't add optimization flags by liquidSVM which may not be known e.g. on PPC
```
Comment (boegel) [inline easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb:2135]: @terjekv This should go right above the `'preinstallopts': 'LIQUIDSVM_TARGET="empty"',` line (so maybe move that down again)?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb (hunk 5 above)
Comment (terjekv) [reply easybuild/easyconfigs/r/R/R-4.0.0-foss-2020a.eb:2135]: Ah, hm, right. Sorted that now. Not *quite* sure what I was thinking at the time.

**Comment by boegel** [comment]
@terjekv Is a test report from `aarch64` on its way?
@branfosj Could you submit a test report from POWER for this?

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node3404.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/cf18bcc22d3ae27bfc0087ae04c8b529 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
bear-pg0305u03a.bear.cluster - Linux RHEL 7.6, POWER, 8335-GTX (power9le), Python 3.6.8
See https://gist.github.com/cc4cb8a741b1056db402244ae6b4f7f1 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3100.skitty.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/8cf5286939aa09341e94c392cbd83a00 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3404.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/2271e8fb88b649b74126a1a892015e33 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2616.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/d6554c1bbc1109d962e34c929be940bd for a full test report.

**Comment by boegel** [comment]
Going in, thanks @terjekv!

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
arm2 - Linux ubuntu 18.04, AArch64, UNKNOWN, Python 3.6.9
See https://gist.github.com/53836b8cb63d8666f7a97d3d623bc622 for a full test report.

**Pull request merged**